### PR TITLE
Correctly propagating exceptions out of the test_start or test_end hooks

### PR DESCRIPTION
**Pull request by pierreluctg**
Restoring the behavior described in the **note** section here https://slash.readthedocs.io/en/master/fixtures.html#test-start-end-for-widely-scoped-fixtures

> Note: 
Exceptions propagating out of the test_start or test_end hooks will fail the test, possibly preventing it from starting properly

### CODE REVIEW

File: tests/test_fixture_start_end_test.py [hunk 1]
```diff
@@ -97,34 +91,28 @@ def __code__():
...
         class SomeTest(slash.Test):
             def before(self, fixture, fixture_failing):  # pylint: disable=unused-argument,arguments-differ
-                self.data = fixture
+                self.data = 1
 
             @slash.parametrize("param", [10, 20, 30])
-            def test_something(self, param):
-                slash.context.result.data["value"] = param + self.data
-
-    suite_builder.build().run().assert_all(6).exception(ZeroDivisionError).with_data(
```
Comment (vmalloc) [inline tests/test_fixture_start_end_test.py]: Why was this removed? I think it's important to make sure the correct values are passed to each iteration, as this is a very delicate mechanism

Comment (pierreluctg) [reply tests/test_fixture_start_end_test.py]: I agree however in this case (with the fix in this PR) we are not supposed to execute the body of the test method when one of the the fixture `this.test_start` raises a exception. That is why the test method is now `slash.add_error("Not supposed to reach here").mark_fatal()`.

Comment (vmalloc) [reply tests/test_fixture_start_end_test.py]: Oh I see now... That's right -- this behavior makes much more sense.

**Comment by vmalloc** [comment]
@pierreluctg can you please add a changelog entry for this PR?

**Comment by pierreluctg** [comment]
Hi @vmalloc, changelog entry added 🐛

**Comment by vmalloc** [comment]
@pierreluctg thanks!

**Comment by vmalloc** [comment]
Thanks for your contribution

**Comment by pierreluctg** [comment]
Hi @vmalloc, can you please make maintenance/bugfix release with this change?

Thank you

**Comment by vmalloc** [comment]
@pierreluctg oops! that's completely my bad -- I published a release branch but forgot to finalize the release. Will be done in the next few hours. Thanks for the reminder!

**Comment by pierreluctg** [comment]
Thank you @vmalloc
